### DISCUSSION TITLE
Difference between gpt-researcher and gptr-nextjs ?

### DISCUSSION THREAD

**Discussion by djuillard** [assafelovic/gpt-researcher#714] [Q&A]
Sorry for asking such a stupid question.... but I can't figure out why.
I have noticed in the docker compose file that there are 2 services calles gptresearcher and gptr-nextjs. I can access both via the two url:ports and I believe that the nextjs is a react refactor of the former portal. I am wondering whether the two services are equivalent (except front end of course) ? Is the initial gptresearcher service will be discontinued in near future ? 

The initial service works fine on my side (and btw : congrats for this awesome job), but with the next.js, nothing happens when I click on the buttons...

Thanks

**Comment by ElishaKay** [comment]
@djuillard 

Hey & thanks for reaching out.

Please try [running the app like so
](https://github.com/assafelovic/gpt-researcher/tree/master/multi_agents#run-the-nextjs-react-app-with-docker)

What error are you seeing in the network tab on localhost:3000 when you try to press the buttons?

We're just using NextJS for the frontend currently - correct, we plan for it to replace the current vanilla JS frontend.

Regarding future direction: the GPTR Python Server is still the core backend (the Arkenstone, if you may)

![68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f776174747061642d6d656469612d736572766963652f53746f7279496d6167652f53774e6d346773746434636f6f413d3d2d3634363538313937312e313536306533643964396534653938303134313636313637393730362e676966](https://github.com/user-attachments/assets/93d53043-81de-48d6-9f34-6414d11a3b9e)

Since NextJS also provides a Javascript server, we made add features there as we go.

If you [join the discord](https://discord.gg/spBgZmm3Xe), you can view 
[a high-level roadmap we're thinking about](https://discord.com/channels/1127851779011391548/1127851779573420053/1266348690499702878)
